src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,112 @@
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (channel, type: ChannelCategory) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: userTarget?.user?.name || userTarget?.user?.username,
+        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId;
+      channel.targetName = chatName?.name;
+      channel.targetImage = chatName?.image;
+
+      channel.channel = {...channel};
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
+  };
+
+  const sendMessageDM = async (
+    id,
+    source: 'post' | 'comment',
+    channelCategory: ChannelCategory
+  ) => {
+    const initChat =
+      channelCategory === 'ANONYMOUS'
+        ? await initChatFromPostAnon({source, id})
+        : await initChatFromPost({source, id});
+
+    const channel = {
+      ...initChat?.data?.channel,
+      better_channel_members: initChat?.data?.better_channel_members,
+      members: initChat?.data?.better_channel_members
+    };
+
+    if (!channel?.members?.length) {
+      console.info('no members');
+      return;
+    }
+
+    if (channel?.type === 'topics') {
+      console.info('type is topics');
+      return;
+    }
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
+
+      await Promise.all(
+        (channel?.messages || []).map(async (message) => {
+          if (message?.type === 'deleted') return;
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          await chat.save(localDb);
+        })
+      );
+
+      await saveChannelList(channel, channelCategory);
+    } catch (e) {
+      console.log('error on saveChannelData', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `sendMessageDM` function, there are several instances where you're using `console.info` for error handling (lines 70, 75). This is not a good practice as it doesn't provide any feedback to the user or the calling function about the error. Consider throwing an error or returning a specific value to indicate an error occurred.

```diff
-      console.info('no members');
-      return;
+      throw new Error('No members');

-      console.info('type is topics');
-      return;
+      throw new Error('Type is topics');
```

Also, the error handling at the end of this function could be improved in a similar way to `saveChannelList`.

```diff
-      console.log('error on saveChannelData', e);
+      console.error('error on saveChannelData', e);
+      throw e;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->